wandb/sdk/internal/sender.py

```diff
@@ -227,6 +234,20 @@ def send_request_stop_status(self, record):
                 logger.warning("Failed to check stop requested status: %s", e)
         self._result_q.put(result)
 
+    def debounce(self) -> None:
+        config_value_dict = self._config_format(self._consolidated_config)
+        if config_value_dict != self._config_at_last_upsert:
+            # TODO(jhr): check result of upsert_run?
+            if self._run:
+                self._api.upsert_run(
+                    name=self._run.run_id,
+                    config=config_value_dict,
+                    **self._api_settings
+                )
+
+        self._debounced_records = []
```

Comment:
I'm a little confused about whether this is necessary. You don't seem to be reading items off this list anywhere? Maybe it's not necessary because we're updating the inmemory config to be "correct"? If so, then I guess this is a real debounce and not a batch :)